flink-table/flink-table-common/src/main/java/org/apache/flink/table/plan/stats/TableStats.java

```diff
@@ -69,4 +69,17 @@ public TableStats copy() {
 		return copy;
 	}
 
+	public TableStats merge(TableStats other) {
+		Map<String, ColumnStats> colStats = new HashMap<>();
+		HashMap<String, ColumnStats> otherColStats = new HashMap<>(other.colStats);
+		for (Map.Entry<String, ColumnStats> entry : this.colStats.entrySet()) {
+			String col = entry.getKey();
+			ColumnStats stats = entry.getValue();
+			ColumnStats otherStats = otherColStats.remove(col);
+			stats = otherStats == null ? stats : stats.merge(otherStats);
+			colStats.put(col, stats);
+		}
+		colStats.putAll(otherColStats);
+		return new TableStats(this.rowCount + other.rowCount, colStats);
```

Comment:
if `rowCount` is -1 (unknown value), the merged `rowCount` should also be unknown